public/docs/dart/latest/guide/forms.jade

```diff
@@ -0,0 +1,662 @@
+
+:marked
+  This code repeats the `<option>` tag for each power in the list of powers.
+  The `#p` local template variable is a different power in each iteration;
+  we display its name using the interpolation syntax with the double curly braces.
+
+
+.l-main-section
+:marked
+  ## Two-way data binding with ***ng-model**
+  Running the app right now would be disappointing.
+
+figure.image-display
+  img(src="/resources/images/devguide/forms/hf-3.png" alt="Early form with no binding")
+:marked
+  The problem is that we aren't binding to the `Hero` yet.
+  We know how to do that from earlier guides.
+  [Displaying Data](./displaying-data.html) demonstrated
+  showing data onscreen with a property binding.
+  [User Input](./user-input.html) showed how to listen for DOM events with an
+  event binding and how to extract values from components.
+
+  Now we need to display, listen, and extract at the same time.
+
+  We could use the techniques we already know, but
+  instead we'll introduce something new: the `NgModel` directive, which
+  makes binding the form to the model super easy.
+
+  Find the `<input>` tag for Name and update it like this:
+
++makeExample('forms/dart/angular2_forms-3-BINDING/lib/hero_form_component.html', 'ng-model-1', 'lib/hero_form_component.html')(format="linenums:6")
+
+.l-sub-section
+  :marked
+    We added a diagnostic interpolation after the input tag
+    so we can see what we're doing.
+    We left ourselves a note to throw it way when we're done.
+
+:marked
+  Focus on the binding syntax: `[(ng-model)]="..."`.
+
+  If we ran the app right now and started typing in the Name input box,
+  adding and deleting characters, we'd see them appearing and disappearing
+  from the interpolated text.
+  At some point it might look like this.
+figure.image-display
+  img(src="/resources/images/devguide/forms/ng-model-in-action.png" alt="ng-model in action")
+:marked
+  The diagnostic is evidence that values really are flowing from the input box to the model and
+  back again. **That's two-way data binding!**
+
+  Let's add similar `[(ng-model)]` bindings to Alter Ego and Hero Power.
+  We'll ditch the input box binding message
+  and add a new binding at the top to the component's `diagnostic` property.
+  Then we can confirm two-way data binding *to the entire Hero model*.
+
+  After revision, the core of the form should have three `[(ng-model)]` bindings that
+  look much like this:
+
++makeExample('forms/dart/angular2_forms-4-DIAGNOSTIC/lib/hero_form_component.html', 'ng-model-2', 'lib/hero_form_component.html')(format="linenums:4")
+
+:marked
+  If we ran the app right now and changed all the values, it might look like this:
+figure.image-display
+  img(src="images/ng-model-in-action-2.png" alt="ng-model in super action")
+:marked
+  The diagnostic near the top of the form
+  confirms that our changes to the values are reflected in the model.
+
```

Comment:
I like this better (although I changed _display_ to _look_ because _display like this_ sounded weird to me; _display something like this_ might be more correct, but then it's getting wordy).